app/docs/dev/policies/meshaccesslog.md

```diff
@@ -0,0 +1,293 @@
+    - key: "start_time"
+      value: "%START_TIME%"
+    - key: "kuma_mesh"
+      value: "%KUMA_MESH%"
+    - key: 'method'
+      value: '"%REQ(:METHOD)%'
+    - key: "path"
+      value: "%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%"
+    - key: 'protocol'
+      value: '%PROTOCOL%'
+    - key: "response_code"
+      value: "%RESPONSE_CODE%"
+    - key: "response_flags"
+      value: "%RESPONSE_FLAGS%"
+    - key: "bytes_received"
+      value: "%BYTES_RECEIVED%"
+    - key: "bytes_sent"
+      value: "%BYTES_SENT%"
+    - key: "duration_ms"
+      value: "%DURATION%"
+    - key: "upstream_service_time"
+      value: "%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%"
+    - key: 'x_forwarded_for'
+      value: '"%REQ(X-FORWARDED-FOR)%"'
+    - key: 'user_agent'
+      value: '"%REQ(USER-AGENT)%"'
+    - key: 'request_id'
+      value: '"%REQ(X-REQUEST-ID)%"'
+    - key: 'authority'
+      value: '"%REQ(:AUTHORITY)%"'
+    - key: "kuma_source_service"
+      value: "%KUMA_SOURCE_SERVICE%"
+    - key: "kuma_destination_service"
+      value: "%KUMA_DESTINATION_SERVICE%"
+    - key: "kuma_source_address_without_port"
+      value: "%KUMA_SOURCE_ADDRESS_WITHOUT_PORT%"
+    - key: "upstream_host"
+      value: "%UPSTREAM_HOST%"
+```
+
+</details>
+
+
+### Backends
+
+#### TCP
+
+You can configure a `TCP` backend with an address:
+
+```yaml
+backends:
+  - tcp:
+      address: 127.0.0.1:5000
+```
+
+#### File
+
+You can configure a `File` backend with a path:
+
+```yaml
+backends:
+  - file:
+      path: /tmp/access.log
+```
+
+## Examples
+
+{% tabs meshaccesslog useUrlFragment=false %}
+{% tab meshaccesslog Kubernetes %}
+
+Full example:
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshAccessLog
+metadata:
+  name: default
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default # optional, defaults to `default` if unset
+spec:
+  targetRef:
+    kind: Mesh
+  default:
+    backends:
+      - tcp:
+          address: 127.0.0.1:5000
+          format:
+            json:
+              - key: "start_time"
+                value: "%START_TIME%"
+      - file:
+          path: /tmp/access.log
+          format:
+            plain: '[%START_TIME%]'
+```
+
+Apply the configuration with `kubectl apply -f [..]`.
+
+{% endtab %}
+{% tab meshaccesslog Universal %}
+
+Full example:
+```yaml
+type: MeshAccessLog
+name: default
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh
+  default:
+    backends:
+      - tcp:
+          address: 127.0.0.1:5000
+          format:
+            json:
+              - key: "start_time"
+                value: "%START_TIME%"
+      - file:
+          path: /tmp/access.log
+          format:
+            plain: '[%START_TIME%]'
+```
+
+Apply the configuration with `kumactl apply -f [..]` or with the [HTTP API](../../reference/http-api).
+
+{% endtab %}
+{% endtabs %}
+
+## Logging external services
+
+When running Kuma on Kubernetes you can also log the traffic to external services.
+To do it use `MeshService` as a targetRef target.
```

Comment:
If this is true we should have an e2e test. I was under the impression this will go over the pass_through cluster and thus only loggable using `kind: Mesh`

Reply:
I wasn't sure so I asked on slack.